mypy_protobuf/main.py

```diff
@@ -269,8 +269,24 @@ def _write_comments(self, scl: SourceCodeLocation) -> bool:
         if sci_loc.trailing_comments is not None:
             lines.extend(self._break_text(sci_loc.trailing_comments))
 
+        lines = [
+            # Replace backslashes with double-backslashes
+            # to remove anything that would become an escape sequence in a Python string.
+            # Escape triple-quotes that would otherwise end the docstring early.
+            line
+                .replace("\\", "\\\\")
+                .replace('"""', '\\"\\"\\"')
+            for line in lines
+        ]
         if len(lines) == 1:
-            self._write_line('"""{}"""', lines[0])
+            line = lines[0]
+            if line.endswith('"'):
+                # Docstrings are terminated with triple-quotes, so if the documentation itself ends in a quote,
+                # insert some whitespace to separate it from the closing quotes.
+                # This is not necessary with multiline comments
+                # because in that case we always insert a newline before the trailing triple-quotes.
+                line = line + " "
+            self._write_line('"""{}"""', line)
```

Comment:
As per my comment in #296, declaring a comment with `"""` does not make it a docstring, which is a very common misconception.

https://www.python.org/dev/peps/pep-0257/#what-is-a-docstring

If you really want to use anonymous string declarations then I would extract the code here into its own method so you can fully unittest various use cases to ensure they end up doing what you actually expect.

Something that could be relatively safe would be to use `repr()` on the strings you want to use as comments and just use that since they are safely parsable:

```// my comment``` would then become `' my comment'`. And things like

It does mean that your anonymous string declarations now use `'` instead of `"""`, but there is nothing that technically links `"""` to docstring except for the convention in pep-0257 linked above.

My recommendation would be to let comments be comments, as with `#`, since that what they technically are for, and then add a new option for folks that do prefer the anonymous string declarations because their IDE is configured to mask comments (I know some people do).

Honestly I feel we are fixing a user problem for the few (invisible comments due to IDE configuration), with a technical problem (safe string parsing is complicated), where there was no problem for most users (comments just worked).

Reply:
repr seems like a good idea.

I do think that the PEP257 describes
```
String literals occurring immediately after a simple assignment at the top level of a module, class, or __init__ method are called "attribute docstrings".
```
which is what the IDE is leaning on here. I don't think it's a configuration issue - but rather this is what the IDE is looking for by the spec.

Reply:
Let me clarify. I think that adding docstrings is great but it should only be the docstrings. For the other proto comments that are inclined alongside the attributes, they should remain comments and not strings.

Reply:
Do you have concrete examples? Eg in https://github.com/dropbox/mypy-protobuf/blob/main/test/generated/testproto/test_pb2.pyi ?

We write PEP257 docstring comments for

- top of the module
- enum class
- message class
- message getter methods
- extension classes
- service methods
- service classes
- grpc methods
- grpc classes

We write PEP257 attribute docstrings for
- enum values
- message scalar fields

I can't find any comments of the variety you're describing.